projects/igniteui-angular/src/lib/icon/icon.component.ts

```diff
@@ -32,63 +32,190 @@ export class IgxIconComponent implements OnInit {
     @ViewChild('explicitLigature', { read: TemplateRef })
     private explicitLigature: TemplateRef<HTMLElement>;
 
+    /**
+     *  This allows you to disable `igx-icon` class. By default it's applied.
+     *```html
+     *@ViewChild("MyIcon")
+     *public icon: IgxIconComponent;
+     *ngAfterViewInit() {
+     *    icon.cssClass = false;
+     * }
+     * ```
+     */
     @HostBinding('class.igx-icon')
     public cssClass = 'igx-icon';
 
+    /**
+     *  This allows you to disable the `aria-hidden` attribute. By default it's applied.
+     *```html
+     *@ViewChild("MyIcon")
+     *public icon: IgxIconComponent;
+     *ngAfterViewInit() {
+     *    icon.ariaHidden = false;
+     * }
+     * ```
+     */
     @HostBinding('attr.aria-hidden')
     public ariaHidden = true;
 
+    /**
+    *    An @Input property that sets the value of the `id` attribute.
+    *```html
+    *<igx-icon id="igx-icon-1" fontSet="material" name="settings" color="blue" [isActive]="false"></igx-icon>
+    *```
+    */
     @HostBinding('attr.id')
     @Input()
     public id = `igx-icon-${NEXT_ID++}`;
 
+    /**
+    *    An @Input property that sets the value of the `fontSet`. By default it's "material".
+    *```html
+    *<igx-icon fontSet="material" name="settings" color="blue" [isActive]="false"></igx-icon>
+    *```
+    */
     @Input('fontSet')
     public font: string;
 
+    /**
+    *    An @Input property that allows you to disable the `active` property. By default it's applied.
+    *```html
+    *<igx-icon [isActive]="false" fontSet="material" name="settings" color="blue"></igx-icon>
+    *```
+    */
     @Input('isActive')
     public active = true;
 
+    /**
+    *    An @Input property that allows you to change the `iconColor` of the icon.
+    *```html
+    *<igx-icon color="blue" [isActive]="true" fontSet="material" name="settings" ></igx-icon>
+    *```
+    */
     @Input('color')
     public iconColor: string;
 
+    /**
+    *    An @Input property that allows you to change the iconName of the icon.
+    *    You can provide either ligature `name` or glyph `iconName`, but not both at the same time.
+    *```html
+    *<igx-icon name="question_answer" color="blue" [isActive]="true" fontSet="material"></igx-icon>
+    *```
+    */
     @Input('name')
     public iconName: string;
 
+    /**
+    *    An @Input property that allows you to change the `glyphName` of the icon.
+    *    You can provide either ligature `name` or glyph `iconName`, but not both at the same time.
+    *```html
```

Comment:
`iconName`, `name` and `glyphName` are a bit confusing. It should be explicitly described that `glyphName` can be set using `iconName`